debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -194,11 +194,13 @@ private LogMinerEventProcessor createProcessor(ChangeEventSourceContext context,
      * Gets the first system change number in both archive and redo logs.
      *
      * @param connection database connection, should not be {@code null}
+     * @param archiveLogRetention duration that archive logs are mined
+     * @param archiveDestinationName configured archive destination name to use, may be {@code null}
      * @return the oldest system change number
      * @throws SQLException if a database exception occurred
      * @throws DebeziumException if the oldest system change number cannot be found due to no logs available
      */
-    private Scn getFirstScnInLogs(OracleConnection connection) throws SQLException {
+    public static Scn getFirstScnInLogs(OracleConnection connection, Duration archiveLogRetention, String archiveDestinationName) throws SQLException {
```

Comment:
As discussed above, lets avoid making this `public static` and instead move the method to the `OracleConnection` class as a `public` method and access it via the connection implementation instead.

Reply:
Due to abandon verified in `OracleConnectorTask` the redo log position, I submit that paragraph code does revert, I reverted the code commit.